enumerables.rb

```diff
@@ -0,0 +1,152 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    n = 0
+    while n < size
+      yield(self[n])
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    n = 0
+    x = 0
+    while n < size
+      yield(self[n], x)
+      n += 1
+      x += 1
+
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    n = 0
+    outcome = []
+    while n < size
+      outcome.push(self[n]) if yield(self[n])
+      n += 1
+    end
+    outcome
+  end
+
+  def my_all
+    return to_enum(:my_all) unless block_given?
+
+    n = 0
+    x = 0
+    outcome = []
+    while n < size
+      if yield(self[n])
+        outcome.push(true)
+      else
+        outcome.push(false)
+      end
+      n += 1
+    end
+    result = true
+    while x < outcome.size
+      result = false if outcome[x] == false
+      x += 1
+    end
+    result
+  end
```

Comment:
## Range and array self
- [ ] Test with hashes and ranges, then compare with original `all` Ruby method
- [ ] Fix your method to work with the tests. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)

**Example:**

`test_file.rb`
```ruby
(0..9).my_all{
# or
Range.new(0, 9)

```

## Optional argument
- [ ] Add  optional argument to this method. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)


```ruby
def my_all?(argument = nil)
```